js/shared.js

```diff
@@ -378,4 +377,75 @@ async function getAllWitnessesOfSource(source){
  */ 
 function undoBrowserSelection(s){
     s?.removeAllRanges?.() ?? s?.empty?.()
+}
+
+/** 
+ * Confirm the shelfmark's inclusion in "GoG-manuscripts" dynamic collection. 
+ * @see {@link findShelfmark #113} to verify shelfmark is appropriate. 
+ *
+ * @async
+ * @function addManuscriptToGoG
+ * @param { string } shelfmark Identifier to include.
+ */ 
+async function addManuscriptToGoG(shelfmark) {
+    try {
+        /** Wash shelfmark by 
+         * Removing specific special characters @ $ % * ?
+         * Replace multiple spaces with a single space
+         * Removing trailing or leading whitespace
+         * */
+        const cleanShelfmark = shelfmark.replace(/[@$%*?]+/g, '') 
+        .replace(/\s+/g, ' ') 
+        .trim() 
+
+        // check for existing annotation with cleaned shelfmark
+        const query = {
+            "target": `${cleanShelfmark}`,
+        }
+        const response = await fetch(`${__constants.tiny+"/query"}`, {
+            method: 'POST',
+            mode: 'cors',
+            headers: {
+                "Content-Type": "application/json;charset=utf-8"
+            },
+            body: JSON.stringify(query)
+        })
+        if (!response.ok) throw new Error('Network response was not ok.')
+
+        const existingAnnotations = await response.json();
+
+        console.log(existingAnnotations)       
+        if(existingAnnotations.length > 0){
+            const ev = new CustomEvent("Annotation already exists.")
+            globalFeedbackBlip(ev, 'Annotation already exists for this shelfmark.', false)
+            return
+        }
+
+        // save new annotation with shelfmark if none exists
+        const annotation = {
+            "@context": "http://www.w3.org/ns/anno.jsonld",
+            "@type": "Annotation",
+            "target": `${cleanShelfmark}`,
+            "body": { "targetCollection": "GoG-manuscripts" }
+        }
+        const saveResponse = await fetch(`${__constants.tiny+"/create"}`, { 
```

Comment:
Same here, change this around a little bit so you end up with the JSON for the object you saved or your handled error if things go wrong.  You can control what this variable becomes by returning something when you handle the error, such as `null` like `catch(err => {return null})`. 

`const savedResponse = fetch().then(resp=>resp.json()).catch(err => {return new Error("AHHH")})`